website/static/js/home-page/newAndNoteworthyPlugin.js

```diff
@@ -0,0 +1,166 @@
+        var popularPromise = m.request({method: 'GET', url: popularUrl, config: xhrconfig, background: true});
+        popularPromise.then(function(result){
+            var numPopular = Math.min(result.data.length, self.SHOW_TOTAL);
+            for (var l = 0; l < numPopular; l++) {
+                self.popularNodes().push(result.data[l]);
+                self.fetchContributors(result.data[l]);
+            }
+            self.someDataLoaded(true);
+        }, function _error(result){
+            self.requestError(result);
+        });
+
+        // Additional API call to fetch node link contributors
+        self.fetchContributors = function(nodeLink) {
+            var url = nodeLink.embeds.target_node.data.relationships.contributors.links.related.href;
+            var promise = m.request({method: 'GET', url : url, config: xhrconfig});
+            promise.then(function(result){
+                var contribNames = [];
+                result.data.forEach(function (contrib){
+                    contribNames.push($osf.findContribName(contrib.embeds.users.data.attributes));
+                });
+                var numContrib = result.links.meta.total;
+                var nodeId = nodeLink.id;
+                self.contributorsMapping[nodeId] = [contribNames, numContrib];
+            }, function _error(result){
+                self.requestError(result);
+            });
+        };
+
+        // Gets contrib family name for display
+        self.getFamilyName = function(i, node) {
+            return self.contributorsMapping[node.id][0][i];
+        };
+
+        self.addToolTip = function(line) {
+            var $line = $(line);
+            if (line.offsetWidth < line.scrollWidth) {
+                $line.tooltip('show');
+            }
+        };
+    },
+    view : function(ctrl) {
+        if (ctrl.errorLoading()) {
+            return m('p.text-center.m-v-lg', 'Error loading projects. Please refresh the page.');
+        }
+
+        if (!ctrl.someDataLoaded()) {
+            return m('.text-center', m('.logo-spin.logo-xl.m-v-xl'));
+        }
+
+        function newAndNoteworthyProjectsTemplate () {
+            return ctrl.newAndNoteworthyNodes().map(function(node){
+                return m.component(NoteworthyNodeDisplay, {
+                    node : node,
+                    getFamilyName: ctrl.getFamilyName,
+                    addToolTip: function(item) {
+                        return ctrl.addToolTip(item);
+                    },
+                    contributorsMapping: ctrl.contributorsMapping
+                });
+            });
+        }
+
+        function popularProjectsTemplate () {
+            return ctrl.popularNodes().map(function(node){
+                return m.component(NoteworthyNodeDisplay, {
+                    node : node,
+                    getFamilyName: ctrl.getFamilyName,
+                    addToolTip: function(item) {
```

Comment:
If the tooltip-adding logic will never vary across usages of `NoteWorthyNodeDisplay`, it doesn't need to be an arg.